File: src/main/java/org/vladg/ast/repository/model/annotation/AnnotatedEntity.java
package org.vladg.ast.repository.model.annotation;

import org.vladg.ast.repository.model.ASTObject;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

public abstract class AnnotatedEntity extends ASTObject {

    @Getter
    public List<Annotation> annotationList = new ArrayList<>();

    public void addAnnotation(Annotation annotation) {
        this.annotationList.add(annotation);
    }

}

File: src/main/java/org/vladg/ast/repository/model/annotation/AnnotatedEntity.kt
package org.vladg.ast.repository.model.annotation

import org.vladg.ast.repository.model.ASTObject


abstract class AnnotatedEntity : ASTObject() {

    var annotationList: MutableList<Annotation> = ArrayList()

    fun addAnnotation(annotation: Annotation) {
        annotationList.add(annotation)
    }

}